__tests__/validations/component.test.tsx

```diff
@@ -115,13 +115,104 @@ test('Throw when one of the prefabs options is invalid', (t: Context): void => {
         },
       ],
     },
-  ] as unknown) as Prefab[];
+  ] as unknown as Prefab[];
+
+  t.throws(() => validatePrefabs(prefabs));
+});
+
+test('Throw when type partial has descendants', (t: Context): void => {
+  const prefabs = [
+    {
+      name: 'Component Name',
+      icon: 'TitleIcon',
+      category: 'CONTENT',
+      structure: [
+        {
+          name: 'something',
+          options: [
+            {
+              value: '',
+              label: 'something',
+              key: 'something',
+              type: 'TEXT',
+            },
+          ],
+          descendants: [
+            {
```

Comment:
I don't understand this part. Why did we step away from:

```typescript
descendants: [
  {
    type: "PARTIAL",
    id: "UUID"
  }
]
```

Reply:
we have changed the partial object the way you mentioned. There are also testes included for this.